### DISCUSSION TITLE
Help on typing custom FilterFns

### DISCUSSION THREAD

**Discussion by mgrog** [TanStack/table#4554] [Q&A]
Hey everyone. This library has been an incredible time saver for me, but I've run against an issue I can't find a great solution for.

In the example for [filtering a table](https://tanstack.com/table/v8/docs/examples/react/filters) what appears to be the recommended way of adding custom FilterFns in typescript is to use a variation of the following code: 
```
declare module '@tanstack/table-core' {
  interface FilterFns {
    fuzzy: FilterFn<unknown>
  }
}
```

This works well when I have a single react table instance in my code base, but when I have another instance I find the `useReactTable` hook now globally requires a filterFns option with those same custom filters defined, even when I have no use for those custom filters in another table.

After some searching I have given up and type the custom filter as `any` in my `ColumnDef`'s to be done with it. This is obviously not ideal for type safety, so I'm wondering has anyone else encountered this issue? Are there any alternatives to declare module for table-core? I've tried looking into the source code but can't find any obvious way of doing it in locally scoped way.

Thanks in advance to all who might help me find a better solution.

**Comment by fn0rth** [comment]
The problem comes from https://github.com/TanStack/table/blob/ab2819cb53098e1891d68041e439692afc2d52b7/packages/table-core/src/features/ColumnFiltering.ts#L189-L195 as soon as we add something to `FilterFns` it'll be not `never` and `filterFns` will be required. I wonder why it is enforced once augmented?

A possible solution is to augment a bit further by overwriting `ColumnFiltersOptions`
```ts
declare module '@tanstack/table-core' {
  interface FilterFns {
    fuzzy: FilterFn<unknown>;
  }
  interface ColumnFiltersOptions<TData extends RowData> {
    filterFns?: Record<string, FilterFn<TData>>;
  }
}
```
